lab_5_scrapper/scrapper.py

```diff
@@ -140,10 +298,79 @@ def get_search_urls(self) -> list:
         Returns:
             list: seed_urls param
         """
+        return self.config.get_seed_urls()
 
 
 # 10
 # 4, 6, 8, 10
+class CrawlerRecursive(Crawler):
+    """
+    Recursive Crawler is a child of Crawler class.
+    """
+    def __init__(self, config: Config) -> None:
+        super().__init__(config)
+
+        with open('already_crawled_urls.txt', 'r', encoding='utf-8') as f:
+            self.already_crawled = f.readlines()
+
+        self.article_tag = 'ARTICLE URL: '
+        self.urls = [url[len(self.article_tag):-1:]
+                     for url in self.already_crawled
+                     if self.article_tag in url]
+
+        if self.already_crawled:
+            self.start_url = self.urls[-1]
+            # to not encounter core element url
+        else:
+            self.start_url = self.base_url
+
+    def find_articles(self) -> None:
+        response = make_request(self.start_url, self.config)
+        if not response.ok:
+            return
+
+        article_bs = BeautifulSoup(response.text, "lxml")
+
+        all_urls = [self.start_url]
+        all_urls.extend(self.base_url[:-len('/news'):] + link.get('href')
+                        for link in article_bs.find_all(href=True)
+                        if 'http' not in link.get('href'))
+        # all urls to my site pages are put into the html code as
+        # /something/..., but not the full link like
+        # https://www.sbras.info/news/something/...,
+        # so everything that refers to other https pages will break requests.get
+        # because it will be https://www.sbras.infohttp/...
+        # also, there are lots of links that are not on the page,
+        # but refer to core components, like .ico/.css files, so there's no need to have them
+        # but they are handled by request generally
+
+        for url in all_urls:
+            if url in self.already_crawled and url != self.base_url:
+                continue
+            elif url != self.base_url:
+                self.already_crawled.append(url)
+            self.start_url = url
+
+            extracted_url = self._extract_url(article_bs)
+            while extracted_url:
+                if extracted_url in self.urls:
+                    extracted_url = self._extract_url(article_bs)
+                    continue
+
+                self.urls.append(extracted_url)
+                self.already_crawled.append(self.article_tag + extracted_url)
+
+                if len(self.urls) == self.config.get_num_articles():
+                    raise EnoughArticlesException
```

Comment:
reaching the target is an exception? it is conceptually wrong. rework

Reply:
Okay

Reply:
Ok, reworked as return conditions